### ISSUE TITLE
[V3] Implement new Concept Exercise: Functions

### ISSUE THREAD

**Issue by junedev** [exercism/javascript#994]
## Getting started
**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)
- [What are concepts and how they are structured?](https://github.com/exercism/docs/blob/main/anatomy/tracks/concepts.md)
- [What are concept exercises and how they are structured?](https://github.com/exercism/docs/blob/main/anatomy/tracks/concept-exercises.md)

Please also watch the following video:
- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

See the documentation _above_ (general documentation), as well as [How to implement a Concept Exercise in JavaScript](https://github.com/exercism/javascript/blob/main/reference/implementing-a-concept-exercise.md).

## Goal
Functions were already briefly mentioned in the [basics concept](https://github.com/exercism/javascript/blob/main/concepts/basics/introduction.md). The goal of this exercise is to give a proper introduction to functions so the student is prepared to get introduced to callbacks in a later concept exercise.

## Concepts
The following concept page need to be created. You can use the same content for the introduction.md of the concept and the concept exercise. No need to create different content at this point.
* `functions`

## Learning Objectives
In the concept/introduction the student should learn about the following things and then practice them in the concept exercise.

* How to define a function (it is probably ok to repeat the basics in some more detail here)
* All arguments are optional in JS
* What happens if the function is called with fewer or more arguments than mentioned in the function definition
* How to set default parameters
* Return an object if you want to return multiple values
* Pass by value vs. pass by reference
* Functions are first-class objects (can be assigned to variables, passed around etc.)
* [Function Scope](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions#function_scope)
* What happens if there is no return statement

## Out of Scope
The following topics will be introduced later and should therefore not be part of this concept exercise.

* Rest parameters
* Callbacks
* Methods
* Arrow functions (will be introduced later together with callbacks)
* Recursion
* Closures
* Higher Order Functions

## About.md of the Concept
Here some ideas of what could be mentioned.

* Anonymous functions
* Talk about hoisting, explain that function declarations are hoisted, function expressions are not
* [Arguments array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions#using_the_arguments_object)

## Prerequisites
* `objects`
* `null-undefined`

See [this list](https://github.com/exercism/javascript/issues/951#issuecomment-778834953) for details on the learning curve we are aiming for.

## Exercise Idea
This concept is very special to JS so there is probably no fitting exercise in other tracks to use as a template and the exercise needs to be created from scratch. As general inspiration for the story, [Lasanga Master](https://github.com/exercism/swift/blob/main/exercises/concept/lasagna-master/.docs/instructions.md) might help.

## Help
You can choose to do this solo-style, or collaborate with multiple people on this. The suggested approach is to 
1. **First accept this issue by saying "I'd like to work on this"** (you will not get a response, just go with it) and optionally request that someone works with you (and wait for a second person to accept your request).
2. Use this issue to write down what you need (collect reference material) and discuss, if necessary, what to write and what not to write.
3. Draft a PR and request the feedback from at least one other JavaScript member (you can ping the #maintaining-javascript channel in Slack).

**Comment by junedev** [comment]
I will work on this but it will probably take a while as there is quite some ground to cover and exercise tasks to invent.
